lib/widgets/dialog.dart

```diff
@@ -15,16 +15,33 @@ Widget _dialogActionText(String text) {
   );
 }
 
-/// Displays an [AlertDialog] with a dismiss button.
+/// A wrapper providing access to the status of an [AlertDialog].
 ///
-/// Returns a [Future] that resolves when the dialog is closed.
-Future<void> showErrorDialog({
+/// See also:
+///  * [showDialog], whose return value this class is intended to wrap.
+class DialogStatusController {
+  const DialogStatusController(this._closed);
+
+  final Future<void> _closed;
+  /// Resolves when the dialog is closed.
+  Future<void> get closed => _closed;
```

Comment:
nit: getter before private field: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#order-other-class-members-in-a-way-that-makes-sense